src/Microsoft.Health.Dicom.Core/Features/Validation/ElementMaxLengthValidation.cs

```diff
@@ -35,7 +36,12 @@ public static void Validate(string value, int maxLength, string name, DicomVR vr
             EnsureArg.IsNotNull(vr, nameof(vr));
             if (value?.Length > maxLength)
             {
-                throw ElementValidationExceptionFactory.CreateExceedMaxLengthException(name, vr, value, maxLength);
+                throw new ElementValidationException(
+                    name,
+                    vr,
+                    value,
+                    ValidationErrorCode.ExceedMaxLength,
+                    string.Format(CultureInfo.CurrentCulture, DicomCoreResource.ErrorMessageExceedMaxLength, maxLength));
```

Comment:
Is there a reason we are using `CurrentCulture` at some places and `InvariantCulture` at other?

Reply:
In theory, this should really be `CurrentCulture` as the resources themselves are based on `CurrentUICulture`. But after looking at the BCL libraries on GitHub, they appear to use `CurrentCulture` in these scenarios. I think I incorrectly used `Invariant` in a number of places, but this is mitigated by the fact we are often formatting string args which cannot be localized.